python/lsst/jointcal/jointcal.py

```diff
@@ -614,6 +609,67 @@ def _do_load_refcat_and_fit(self, associations, defaultFilter, center, radius,
 
         return result
 
+    def _load_reference_catalog(self, refObjLoader, center, radius, filterName,
+                                applyColorterms=False):
+        """Load the necessary reference catalog sources, convert fluxes to
+        correct units, and apply color term corrections if requested.
+
+        Parameters
+        ----------
+        refObjLoader : `lsst.meas.algorithms.LoadReferenceObjectsTask`
+            The reference catalog loader to use to get the data.
+        center : `lsst.geom.SpherePoint`
+            The center around which to load sources.
+        radius : `lsst.geom.Angle`
+            The radius around ``center`` to load sources in.
+        filterName : `str`
+            The name of the camera filter to load fluxes for.
+        applyColorterms : `bool`
+            Apply colorterm corrections to the refcat for ``filterName``?
+
+        Returns
+        -------
+        refCat : `lsst.afw.table.SimpleCatalog`
+            The loaded reference catalog.
+        fluxField : `str`
+            The name of the reference catalog flux field appropriate for ``filterName``.
+        """
+        skyCircle = refObjLoader.loadSkyCircle(center,
+                                               afwGeom.Angle(radius, afwGeom.radians),
+                                               filterName)
+
+        # Need memory contiguity to get reference filters as a vector.
+        if not skyCircle.refCat.isContiguous():
+            refCat = skyCircle.refCat.copy(deep=True)
+        else:
+            refCat = skyCircle.refCat
+
+        if applyColorterms:
+            try:
+                refCatName = refObjLoader.ref_dataset_name
+            except AttributeError:
+                # NOTE: we need this try:except: block in place until we've completely removed a.net support.
+                raise RuntimeError("Cannot perform colorterm corrections with a.net refcats.")
+            self.log.info("Applying color terms for filterName=%r reference catalog=%s",
+                          filterName, refCatName)
+            colorterm = self.config.colorterms.getColorterm(
+                filterName=filterName, photoCatName=refCatName, doRaise=True)
+
+            refMag, refMagErr = colorterm.getCorrectedMagnitudes(refCat, filterName)
+            refCat[skyCircle.fluxField] = u.Magnitude(refMag, u.ABmag).to_value(u.nJy)
+            # TODO: I didn't want to use this, but I'll deal with it in DM-16903
+            refCat[skyCircle.fluxField+'Err'] = fluxErrFromABMagErr(refMagErr, refMag) * 1e9
+        else:
+            # TODO: need to scale these until RFC-549 is completed and refcats return nanojansky
```

Comment:
Are these two TODOs related?

Reply:
Yes, although they may not happen at the exact same time.